diff --git a/src/nl/starapple/backend/RunPoker.java b/src/nl/starapple/backend/RunPoker.java
index 759d184..64cd841 100644
--- a/src/nl/starapple/backend/RunPoker.java
+++ b/src/nl/starapple/backend/RunPoker.java
@@ -49,20 +49,30 @@ public class RunPoker
     String bot1Command, bot2Command;
 
 	String playerName1, playerName2;
-	
+
+	int rngSeed;
 	MatchPlayer engine;
     List<PokerBot> bots;
 
 	public static void main(String args[]) throws Exception
 	{	
-		RunPoker run = new RunPoker(args);
-        try {
-		  run.go();
-        }
-        catch(Exception e) {
-            e.printStackTrace();
-            run.finish();
+
+        //for (int i=0;i<10;i++)
+        for (int i=0;i<1;i++)
+        {
+            System.out.println("Round: " + i);
+
+            RunPoker run = new RunPoker(args);
+            try {
+              run.go();
+            }
+            catch(Exception e) {
+                e.printStackTrace();
+                run.finish();
+                System.out.println("Exception in round: " + i);
+            }
         }
+        System.exit(0);
 	}
 
 	public RunPoker(String args[])
@@ -70,6 +80,8 @@ public class RunPoker
 		startingStack = Integer.valueOf(args[0]);
         this.bot1Command = args[1];
         this.bot2Command = args[2];
+        this.rngSeed = Integer.parseInt(args[3]);
+        gameType = Integer.parseInt(args[4]);
 
 		this.playerName1 = "player1";
 		this.playerName2 = "player2";
@@ -91,13 +103,13 @@ public class RunPoker
 		bots.add(player2);
 
 		//gametype is omaha tournament (with two players):
-		gameType = 18;
+		//gameType = 18; // 13 for no limit holdem
 		//number of hands is unlimited
 		numberOfHands = -1;
 		//one winner
 		placesPaid = 1;
 		
-		engine = new MatchPlayer(new ArrayList<PokerBot>(bots), gameType, startingStack);
+		engine = new MatchPlayer(new ArrayList<PokerBot>(bots), gameType, startingStack, rngSeed);
         engine.finishSetup(true);
         ArrayList<String> results = engine.run(numberOfHands, 0);
         for(int i = 0; i < results.size(); i++)
@@ -127,7 +139,7 @@ public class RunPoker
         }
         catch(Exception e) {}
 
-        System.exit(0);
+        //System.exit(0);
 	}
     
 	/**
@@ -288,4 +300,4 @@ public class RunPoker
         System.out.println(bot2.getDump());	
 	}
 
-}
\ No newline at end of file
+}
